src/database/sqlite/migration.py

```diff
@@ -9,274 +9,328 @@
+        """Retrieve the version from the database."""
+
+    @abstractmethod
+    def set_version(self, version: Version):
+        """Set the version field of the database to *version*."""
 
     @property
     @abstractmethod
-    def base_module(self) -> ModuleType:
-        pass
+    def latest_version(self) -> Version:
+        """Latest expected version of the """
+
+    @abstractmethod
+    def get_migration_from_legacy_version(self) -> str | None:
+        """Migrations used to be handled through version number.
+        This method ensures the compatibility between the two systems."""
+        # TODO remove once all legacy migrations have been squashed
+
+    @abstractmethod
+    def remove_legacy_version_field(self):
+        """Remove the version field which was used
+        as the migration field in the legacy migration system.
+        Field might not exist."""
+        # TODO remove once all legacy migrations have been squashed
 
     @property
     def migration_modules(self) -> list[str]:
         return [
-            f'{self.base_module.__name__}.{module}'
-            for module in self._migration_module_names
+            self._get_migration_module_name(migration)
+            for migration in self.migrations
         ]
 
     @cached_property
-    def migration_versions(self) -> list[Version]:
-        return [
-            self._module_name_to_version(module)
-            for module in self._migration_module_names
-        ]
+    def migrations(self) -> list[str]:
+        return sorted([
+            module for _, module, _ in iter_modules(
+                self.base_migration_module.__path__
+            )
+        ])
 
-    @property
-    def _migration_module_names(self) -> list[str]:
-        return [
-            module for _, module, _ in iter_modules(self.base_module.__path__)
-        ]
+    def check_status(self) -> bool:
+        """Check if the database is migrated to the latest migration.
+        Raise a PapiWebException if the stored migration is unknown.
+        Assert that the timeline is valid."""
+        if DEVEL_ENV:
+            self._check_timeline()
+        if not self.is_metadata_installed:
+            self.get_migration_from_legacy_version()
+            return False
 
-    @property
-    def first_migration_version(self) -> Version:
-        return self._ordered_migration_versions[0]
+        version = self.get_version()
+        if version > self.latest_version:
+            raise PapiWebException(
+                f'Database version [{version}] is after current app version '
+                f'{self.latest_version}.'
+            )
+        status = True
+        if version < self.latest_version:
+            status = False
+        migration = self.get_migration()
+        if migration not in self.migrations + [self.MIGRATION_ZERO]:
+            if migration > self.migrations[-1]:
+                message = (
+                    'Database can only be opened by a next version of Papi-Web'
```

Comment:
Could this use `common.APP_NAME`, somehow?
It would make sure we don't have to change all the instances of "Papi-Web" (and variants) when we change name.